packages/react-jsx-highcharts/src/index.js.flow

```diff
@@ -0,0 +1,182 @@
+  from: any,
+  to: any,
+  color?: string,
+}
+type PlotBandLabelProps = {
+  children: Node
+}
+function PlotBandLabel (props: PlotBandLabelProps) { }
+export function PlotBand(props: PlotBandProps) {
+
+}
+PlotBand.Label = PlotBandLabel;
+
+type PlotLineProps = {
+  id?: string,
+  value: any,
+  color?: string,
+}
+type PlotLineLabelProps = {
+  children: Node
+}
+function PlotLineLabel (props: PlotLineLabelProps) { }
+export function PlotLine (props:PlotLineProps) {
+
+}
+PlotLine.Label = PlotLineLabel;
+
+type SubtitleProps = {
+  children: Node
+}
+export function Subtitle (props: SubtitleProps) { }
+
+type TitleProps = {
+  children: Node
+}
+export function Title (props: TitleProps) { }
+
+type TooltipProps = {
+  children: Node,
+  enabled?: boolean
+}
+export function Tooltip (props:TooltipProps) { }
+
+type AxisProps = {
+  type?: 'category' | 'linear' | 'logarithmic' | 'datetime',
+  id?: string,
+  children?: Node
+}
+type AxisTitleProps = {
+  children?: Node
+}
+
+function AxisTitle (props:AxisTitleProps) {
+
+}
+
+export function YAxis (props: AxisProps) {
+
+}
+YAxis.Title = AxisTitle;
+
+export function XAxis (props: AxisProps) {
+
+}
+XAxis.Title = AxisTitle;
+
+export class ZAxis extends React.Component<AxisProps> { }
+
+// this probable needs some refinement
+type DataType = Array<any>
+
+type SeriesProps = {
+  id?: string,
+  data?: DataType,
+  visible?: boolean,
+  children?: Node,
+  className?: string
+}
+export class AreaRangeSeries extends React.Component<SeriesProps> { }
+export class AreaSeries extends React.Component<SeriesProps> { }
+export class AreaSplineRangeSeries extends React.Component<SeriesProps> { }
+export class AreaSplineSeries extends React.Component<SeriesProps> { }
+
+export function BarSeries (props: SeriesProps) { }
+export class BoxPlotSeries extends React.Component<SeriesProps> { }
+export class BubbleSeries extends React.Component<SeriesProps> { }
+export class ColumnRangeSeries extends React.Component<SeriesProps> { }
+export class ColumnSeries extends React.Component<SeriesProps> { }
+export class ErrorBarSeries extends React.Component<SeriesProps> { }
+export class FunnelSeries extends React.Component<SeriesProps> { }
+export class LineSeries extends React.Component<SeriesProps> { }
+export class PieSeries extends React.Component<SeriesProps> { }
+export class PolygonSeries extends React.Component<SeriesProps> { }
+export class PyramidSeries extends React.Component<SeriesProps> { }
+export class SankeySeries extends React.Component<SeriesProps> { }
+export class ScatterSeries extends React.Component<SeriesProps> { }
+export class SplineSeries extends React.Component<SeriesProps> { }
+export class StreamGraphSeries extends React.Component<SeriesProps> { }
+export class TreemapSeries extends React.Component<SeriesProps> { }
+export class WaterfallSeries extends React.Component<SeriesProps> { }
+
+export function withHighcharts (Component: ComponentType<*>, Highcharts: HighchartsType) { }
```

Comment:
Could we also export 

```js
export function provideChart (Component: ComponentType<*>) { }
export function provideAxis (Component: ComponentType<*>) { }
export function provideSeries (Component: ComponentType<*>) { }
```

Reply:
I will add these exports too. Originally I thought them to be internal detail, but they can actually be used to make plugins for this library.

Reply:
Had a looks myself but I can't figure it out. I don't get this Flow type thing at all, nevermind typing a HOC